### DISCUSSION TITLE
Mixed bsp/stacked layout?

### DISCUSSION THREAD

**Discussion by dmitrym0** [asmvik/yabai#1472] [Q&A] [answered]
Hey folks, 

Is it possible to have two regions split vertically where one is a stack and the other is a just an individual window? It seems like it should be but I can't find a reference to this.

**Comment by cherche** [comment, marked as answer]
Yes, this is possible. Call `yabai -m window --insert stack` while focused on the window which you want into make a stack. Then you may add a window to this insertion point as usual (either by warping or opening a new window while the insertion point overlay is active).

I don't know if there's a way to make all newly opened windows automatically add to this stack.

**Reply by dmitrym0** [reply]
>  Then you may add a window to this insertion point as usual (either by warping or opening a new window while the insertion point overlay is active).

I'm having a bit of an issue with this step. So when `yabai -m window --insert stack` my window now has a red overlay. What should be my next step? Warping doesn't seem work for me -- the windows are shuffled.

**Reply by danielo515** [reply]
I am also a bit puzzled with this. Do you get an answer of how you can 
1. Start a stack
2. Add current window to that stack?

**Reply by qaisjp** [reply]
`yabai -m window --insert stack`  puts the window you currently have focused into "stacking mode" (i.e. with an overlay).

Then, focus a different window as normal. Then move that window into the location of the overlay window.

This adds the window you were moving around into the stack of the red overlay window.

For example, say you have Chrome on the left, Firefox on the right. Here's how you make the Chrome window into a "browser" stack:

- Focus Chrome window
- Turn Chrome window into stacking mode: `yabai -m window --insert stack` (you now see an overlay)
- Focus Firefox: `yabai -m window --focus east`
- Move Firefox left (on top of the Chrome window): `yabai -m window --warp west`
- The Firefox window now sits on top of the Chrome window ("browser" stack)
- `yabai -m window --focus stack.next` to change back to Chrome

**Comment by bangedorrunt** [comment]
@cherche and @dmitrym0 you can make window automatically add to stack by using yabai signal:
```bash
# Signals
# Auto stacking windows
yabai -m signal --add event=window_focused action="yabai -m window $YABAI_WINDOW_ID --insert stack"
yabai -m signal --add event=application_front_switched action='yabai -m window --insert stack'
yabai -m signal --add event=window_destroyed action='yabai -m window --insert stack'
yabai -m signal --add event=application_terminated action='yabai -m window --insert stack'
```
Updated: Credit to @dominiklohmann ~or other, really sorry if I couldn't remember coz I got this for a few years and still working like a charm~ and @alextremblay
Ref: #674

**Reply by danielo515** [reply]
So this will add any window focused or opened to the stack, right?

**Reply by Bellavene** [reply]
```
yabai -m signal --add event=window_created action='yabai -m window last --insert stack;SELWIN=$(yabai -m query --windows --window last | jq -r -e ".id");yabai -m window --focus $SELWIN'
```

But still would be great if it would be native.

**Reply by anandkumar89** [reply]
is it possible to stack to north/south/east/west just as warp?

**Reply by Bellavene** [reply]
`yabai -m window north --insert stack; yabai -m window --warp north`
